sdv/single_table/base.py

```diff
@@ -492,7 +492,14 @@ def load(cls, filepath):
                 The loaded synthesizer.
         """
         with open(filepath, 'rb') as f:
-            synthesizer = cloudpickle.load(f)
+            try:
+                synthesizer = cloudpickle.load(f)
+            except RuntimeError:
+                raise SamplingError(
+                    'This synthesizer was created on a machine with GPU but the current machine is'
+                    ' CPU-only. This feature is currently unsupported. We recommend sampling on '
+                    'the same GPU-enabled machine.'
+                )
```

Comment:
Shouldn't this occur on the `sampling` instead ? or the error gets raised while you load the synthesizer?

Reply:
From the [user issue](https://github.com/sdv-dev/SDV/issues/1778), they ran into this problem when loading the synthesizer, so that's where I added the warning.